interface.js

```diff
@@ -0,0 +1,77 @@
+let newsFeed = new NewsFeed();
+
+function getArticles() {
+  fetch('http://news-summary-api.herokuapp.com/guardian?apiRequestUrl=http://content.guardianapis.com/search?show-fields=all')
+    .then(
+      function(response) {
+        response.json().then(function(data) {
+          data.response.results.forEach(function(result) {
+            let title = `${result.webTitle}`
+            let text = `${result.fields.bodyText}`
+            let thumbnail = `${result.fields.thumbnail}`
+            let url = `${result.webUrl}`
+            let newArticle = new Article(title, text, thumbnail, url);
+            newsFeed.addNewArticle(newArticle);
+            let articlesDiv = document.getElementById("article-list")
+            articlesDiv.insertAdjacentHTML('beforeend', `<img class="${newArticle.getUrl()}"src="${newArticle.getThumbnail()}"><br>`)
+            articlesDiv.insertAdjacentHTML('beforeend', `<a class="headline" id="${newArticle.getUrl()}" href="#${newArticle.getUrl()}">${newArticle.getHeadline()}</a><br><br><br>`)
+          })
+        });
+      })
+}
+getArticles()
+
+window.addEventListener("hashchange", function() {
+  hideMainFeed();
+  displayBackButton();
+  let currentHash = getCurrentHash();
+  let article = newsFeed.articles.filter(function(article) {
+    return article.url === currentHash;
+  })
+  let articlesDiv = document.getElementById("article-summary")
+  articlesDiv.insertAdjacentHTML('beforeend', `<img src="${article[0].getThumbnail()}"><br>`)
+  articlesDiv.insertAdjacentHTML('beforeend', `<a target="_blank" rel="noopener noreferrer" href="${article[0].getUrl()}">${article[0].getHeadline()}</a><br>`)
+  articlesDiv.insertAdjacentHTML('beforeend', `<p>${article[0].textBody}</p><br>`)
+})
+
+let backButton = document.getElementById('back-button')
+backButton.addEventListener("click", function() {
+    history.replaceState(null, null, " ");
+    displayNewsFeed();
+    document.getElementById('article-summary').innerHTML = "";
+    hideBackButton()
+})
+
+function hideMainFeed() {
+  var hidden = document.getElementById("article-list")
+  hidden.style.display = "none";
+}
+
+function displayBackButton() {
+  var homeButton = document.getElementById('back-button')
+  homeButton.style.display = (status !== 'open') ? 'block' : 'none';
+}
+
+function getCurrentHash() {
+  return window.location.hash.split("#")[1]
+}
+
+function displayNewsFeed() {
+  var hidden = document.querySelector('#article-list');
+  hidden.style.display = (status !== 'open') ? 'block' : 'none';
+}
+
+function hideBackButton() {
```

Comment:
everything seems to be clear and obvious what each part is doing. Not sure if I can give anything helpful. In  terms of syntax I know that JS adds semicolons after the fact, however adding where appropriate consistently will help readability.

Reply:
added